src/iuse.cpp

```diff
@@ -9187,6 +9187,35 @@ int iuse::disassemble( player *p, item *it, bool, const tripoint & )
     return 0;
 }
 
+int iuse::gobag( player *p, item *it, bool, const tripoint & )
+{
+    auto items = item_group::items_from( "gobag_contents", calendar::turn );
+    bool content_fits = it->typeId() == "personal_gobag";
+    std::string last_armor;
+
+    p->add_msg_if_player( _( "You empty the contents of the go bag onto the floor." ) );
+
+    for( auto &content : items ) {
+        if( content.is_armor() ) {
+            if( content.typeId() != last_armor ) {
+                content_fits = it->typeId() == "personal_gobag" || content.has_flag( "FIT" ) ;
+                last_armor = content.typeId();
+            }
+            content_fits ? content.set_flag( "FIT" ) : content.unset_flag( "FIT" );
```

Comment:
I don't understand the logic for this. Firstly check for `id` of the `gobag_contents` item group. Here goes the duffelbag from this item group. It doesn't have the `FIT` flag. And on the next step this line unsets its already missing `FIT` flag.

Reply:
The game does a random check to see if the item fits when it is created using items_from(). Without any input from me, the game generates items that either randomly fit or do not fit. That check is to insure that all items of the same type (socks, t-shirts) fit the same way. It's a hack but I can't think of a better way to get consistency for the items.

Reply:
I get what you want to do, but I don't get why you're setting the already applied or unsetting already missing flags. If item fits, there's no need to make it fit one more time.

Reply:
Ah, now I get it. I'll make a better if-else statement to handle that.